### DISCUSSION TITLE
Chronology log

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#1111] [Ideas]
I would find it very useful to see a chronology of what I have created and modified in a mindmap.

I anticipate this could be a huge undertaking in terms of development, so it is definitely in the nice-to-have category. But to be able to see the nodes that I created or modified in some way along a simple timeline would be huge. Even showing something like, "Details changed for node #xxx with text: this was the time that..." (showing, perhaps, the node's first 25 characters).

The main use case for this is that in a very large mindmap (mine consists of more than 13,000 nodes), there are days where I am moving in many different parts of the mindmap. The flurry of activity remains in my mind, but not the details. So seeing a day where there was a flurry of activity would let me very quickly focus on that day and find what I remember working on, even a year later.

~ Bal

**Comment by macmarrum** [comment]
Here's my idea of how to aid in identifying nodes modified or created on a particular day.

A script that creates a timeline node, containing modification-date nodes (or creation-date nodes). Each modification-date node contains `script1` which, when executed, instructs Freeplane to apply a filter to show nodes modified on that day.

The script is located (by default) in `Mac2->Create Modification Timeline`.

An analogous script for createdAt: `Mac2->Create Creation Timeline`.

I run the script to create the timeline, then select a timeline node for the desired date and execute its `script1`, e.g. by `Tools->Execute selected node scripts`.
_Note: By default, descendants are hidden; they can be enabled from the Filter toolbar_

- [createModificationTimeline.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/createModificationTimeline.groovy)
- [createCreationTimeline.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/createCreationTimeline.groovy)

_Remove `="/menu_bar/Mac2"` from the top of each script to have them appear in `Tools->Scripts`_
